sdk/keyvault/azure-security-keyvault-certificates/src/test/java/com/azure/security/keyvault/certificates/CertificateClientTestBase.java

```diff
@@ -613,14 +615,9 @@ public void sleep(long millis) {
     static Stream<Arguments> getTestParameters() {
         // when this issues is closed, the newer version of junit will have better support for
         // cartesian product of arguments - https://github.com/junit-team/junit5/issues/1427
-        List<Arguments> argumentsList = new ArrayList<>();
-        getHttpClients()
-            .forEach(httpClient -> {
-                Arrays.stream(CertificateServiceVersion.values()).filter(
-                    CertificateClientTestBase::shouldServiceVersionBeTested)
-                    .forEach(serviceVersion -> argumentsList.add(Arguments.of(httpClient, serviceVersion)));
-            });
-        return argumentsList.stream();
+        List<ServiceVersion> serviceVersions = Arrays.stream(CertificateServiceVersion.values())
+            .filter(CertificateClientTestBase::shouldServiceVersionBeTested).collect(Collectors.toList());
+        return getArgumentsFromServiceVersion(serviceVersions, SERVICE_VERSION_FROM_ENV);
```

Comment:
I still feel like this could be made simpler by moving some of the verbosity up into the getArgumentsFromServiceVersion method, e.g. by calling it like this: `getArgumentsFromServiceVersion(CertificateServiceVersion, CertificateClientTestBase::shouldServiceVersionBeTested, SERVICE_VERSION_FROM_ENV)`

Reply:
Done.